lib/src/drawConfigContext.dart

```diff
@@ -87,31 +126,68 @@ class DrawConfigContext {
     if (primaryFile.exists() == false) {
       primaryFile = new File(this._globalConfigPath.toString());
     }
+  }
 
-    //Parse the ini file.
-    _customConfig = new Config.fromStrings(primaryFile.readAsLinesSync());
-    _initializeAttributes();
+  void _fieldInitializer(type, params) {
+    params.forEach((value) => _initializeField(type, value));
   }
 
   /// Initialize the attributes of the configuration object using the ini file.
-  void _initializeAttributes() {
-    _shortURL = _fetchDefault('short_url');
-    checkForUpdates = _configBool(_fetchDefault('check_for_updates'));
-
-    oauthUrl = _fetch('oauth_url');
-    redditUrl = _fetch('reddit_url');
-
-    //The use of null aware operators here is to give highest precedence to
-    //passed in values to the constructor.
-    clientId ??= _fetchOrNotSet('client_id');
-    clientSecret ??= _fetchOrNotSet('client_secret');
-    httpProxy ??= _fetchOrNotSet('http_proxy');
-    httpsProxy ??= _fetchOrNotSet('https_proxy');
-    _redirectUri ??= _fetchOrNotSet('redirect_uri');
-    refreshToken ??= _fetchOrNotSet('refresh_token');
-    password ??= _fetchOrNotSet('password');
-    userAgent ??= _fetchOrNotSet('user_agent');
-    username ??= _fetchOrNotSet('username');
+  void _initializeField(String type, String param) {
```

Comment:
Any reason we don't just do this all in the constructor? Required fields should always be present and we should throw an exception if they're not. For optional fields with defaults, we can do something like `value = _fetch(field) ?? defaultValue;`, assuming `_fetch` returns `null` if it can't find the field.